src/CoreWCF.BuildTools/src/OperationParameterInjectionGenerator.cs

```diff
@@ -0,0 +1,308 @@
+                        if(implementationCandidates.Length == 1)
+                        {
+                            if (serviceImplAndContract.isPartialClass)
+                            {
+                                GenerateOperationContractImplementation(context, serviceImplAndContract.service, serviceImplAndContract.contract, implementationCandidates[0], operationContract);
+                            }
+                            else if (!isErrorDiagnosticIssued)
+                            {
+                                context.ReportDiagnostic(DiagnosticDescriptors.ServicesShouldBePartialError(serviceImplAndContract.service.Name, serviceImplAndContract.contract.Name));
+                                isErrorDiagnosticIssued = true;
+                            }
+                        }
+                        // TODO multiple candidates..
+                    }
+                    // TODO service is already implemented. should we warn when we detect a candidate
+                }
+            }
+        }
+
+        private static bool GetInstanceContextMode(GeneratorExecutionContext context, INamedTypeSymbol service)
+        {
+            var CoreWCFServiceBehaviorSymbol = context.Compilation.GetTypeByMetadataName("CoreWCF.ServiceBehaviorAttribute");
+            var SSMServiceBehaviorSymbol = context.Compilation.GetTypeByMetadataName("System.ServiceModel.ServiceBehaviorAttribute");
+            
+
+            var serviceBehaviorAttribute = service.GetAttributes()
+                .Where(x => SymbolEqualityComparer.Default.Equals(x.AttributeClass, CoreWCFServiceBehaviorSymbol)
+                    || SymbolEqualityComparer.Default.Equals(x.AttributeClass, SSMServiceBehaviorSymbol))
+                .FirstOrDefault();
+
+            if(serviceBehaviorAttribute == null)
+            {
+                return false;
+            }
+
+            foreach (var argument in serviceBehaviorAttribute.NamedArguments)
+            {
+                if (argument.Key == "InstanceContextMode" && !argument.Value.IsNull)
+                {
+                    return argument.Value.Value.ToString() == "2";
+                }
+            }
+
+            return false;
+        }
+
+        private static void GenerateOperationContractImplementation(GeneratorExecutionContext context, INamedTypeSymbol service, INamedTypeSymbol contract, IMethodSymbol operationContractImplementation, IMethodSymbol operationContract)
+        {
+            var SSMServiceBehaviorSymbol = context.Compilation.GetTypeByMetadataName("System.ServiceModel.ServiceBehaviorAttribute");
+            var CoreWCFServiceBehaviorSymbol = context.Compilation.GetTypeByMetadataName("CoreWCF.ServiceBehaviorAttribute");
+            var TaskSymbol = context.Compilation.GetTypeByMetadataName("System.Threading.Tasks.Task");
+            var GenericTaskSymbol = context.Compilation.GetTypeByMetadataName("System.Threading.Tasks.Task<object>");
+
```

Comment:
This doesn't look to be used anywhere. But should it be `Task<>` for an open generic? It looks like you worked around this by looking at the base type of the return type to see if it's Task. Normally this would be fragile (as Task isn't sealed), but we're constrained to what CoreWCF supports and it only supports Task and Task\<TResult\> task based return types.

Reply:
I finally found the correct syntax which actually is 
```csharp
context.Compilation.GetTypeByMetadataName("System.Threading.Tasks.Task`1")
```

Reply:
and use this symbol to compare with
```
ReturnType.ConstructedFrom
```